packages/misc/src/lib/fetchWithRetry.ts

```diff
@@ -0,0 +1,42 @@
+import fetchRetry, { FetchLibrary, RequestDelayFunction } from 'fetch-retry';
+
+export type RetryDelayHandlerFactory<T extends FetchLibrary> = (
+  interval: number
+) => RequestDelayFunction<T>;
+
+// Intentionally not retrying on `500` since LIT nodes may return 500 in cases where retrying would have negative effect
+export const RETRYABLE_STATUS_CODES = [408, 502, 503, 504];
+
+/** Return a function to be used as `retryDelay` for `fetch-retry`
+ * Retry function uses exponential backoff and jitter to avoid stampeding herd issues
+ *
+ * @param {number} interval How long to use as a base (minimum) retry delay between requests
+ * @returns {RequestDelayFunction} A retry delay handler function for the provided interval
+ */
+export const getRetryDelayHandlerForInterval: RetryDelayHandlerFactory<
+  typeof globalThis.fetch
+> = (interval: number) => {
+  return function getRetryDelay(
+    attempt: number
+    // error: Error | null,
+    // response: Response | null
+  ) {
+    return Math.random() * (Math.pow(2, attempt) * interval);
```

Comment:
Is this truly exponential? since we are multiplying by a random value couldn't we end up with a lower interval?